applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -983,13 +983,24 @@ static void agps_request_handle(struct nrf_modem_gnss_agps_data_frame *incoming_
 	 * to send the request.
 	 */
 #if defined(CONFIG_NRF_CLOUD_AGPS) && defined(CONFIG_NRF_CLOUD_MQTT)
```

Comment:
The logic added here only touches upon the scenario when nRF Cloud MQTT is used as transport. We also need to accommodate for when transports external to nRF Cloud MQTT is used, when `!NRF_CLOUD_MQTT` and `agps_request_encode` is used to encode/handle the A-GPS request.